### ISSUE TITLE
Regarding track by_string

### ISSUE THREAD

**Issue by CyberTaoFlow** [quadrantsec/sagan#1]
Hello,

Can you provide some clarification around the use of "by_string" with after track_by?  It is not really discussed in the documentation but based on how normalization works with field names I would have expected if I had normalized or parsed a JSON or by some other means created a field called "string" then I could track by that arbitrary string.

However looking at the source code of threshold.c and rules.c it becomes apparent that there is no "threshold2_method_string" function but instead this is handled by "threshold2_method_username".

So If I am understanding this correctly:
1. Using "track by_string" is functionally equivalent to "track by_username".
2. Both of which rely on the presence of a "username" field either from JSON parsing or liblognorm.

If the first point is true then I would expect using something like :  

`after: track by_username&by_string ` is basically the same as using either operator alone.

I am looking at how I can use after/threshold to track by two arbitrary fields in a meraki wireless IDS log.  Specifically I want to threshold based on unique pairs of BSSID and SSID......  Being that neither of these is an IP I was hoping that I would be able to combine by_username and a by_string track since by_username is free-form (could use it for SSID) and does not influence the alert header like using src_ip or dst_ip would.

**Comment by beave** [comment]
You are correct.    The idea was the "by_username" was commonly requested but we wanted to make it generic enough to handle things other than just "usernames".  Right now,  you wouldn't be able to do the by_username&by_string because by_string is an alias for by_username.  

It sounds like what we really would want is the ability to search and array of "strings" (which could be usernames, SSIDs, MAC,  whatever).   Maybe drop "by_username" all together as it makes things confusing. 

This probably need to be rethought out.   I'm going to leave this open and put it as a goal for 2.0.1.
